bot/exts/fun/latex/_latex_cog.py

```diff
@@ -0,0 +1,73 @@
+import asyncio
+import hashlib
+import re
+import sys
+from pathlib import Path
+from textwrap import dedent
+
+import discord
+from discord.ext import commands
+
+FORMATTED_CODE_REGEX = re.compile(
+    r"(?P<delim>(?P<block>```)|``?)"        # code delimiter: 1-3 backticks; (?P=block) only matches if it's a block
+    r"(?(block)(?:(?P<lang>[a-z]+)\n)?)"    # if we're in a block, match optional language (only letters plus newline)
+    r"(?:[ \t]*\n)*"                        # any blank (empty or tabs/spaces only) lines before the code
+    r"(?P<code>.*?)"                        # extract all code inside the markup
+    r"\s*"                                  # any more whitespace before the end of the code markup
+    r"(?P=delim)",                          # match the exact same delimiter from the start again
+    re.DOTALL | re.IGNORECASE,              # "." also matches newlines, case insensitive
+)
+
+THIS_DIR = Path(__file__).parent
+CACHE_DIRECTORY = THIS_DIR / "cache"
+CACHE_DIRECTORY.mkdir(exist_ok=True)
+
+
+def _prepare_input(text: str) -> str:
+    text = text.replace(r"\\", "$\n$")  # matplotlib uses \n for newlines, not \\
+
+    if match := FORMATTED_CODE_REGEX.match(text):
+        return match.group("code")
+    else:
+        return text
+
+
+def _format_err(text: str) -> str:
+    # prevent escaping the codeblock by inserting a zero-width-joiner
+    text = text.replace("`", "`\u200d")
+    return dedent(
+        f"""
+        ```
+        {text}
+        ```
+        """
+    ).strip()
+
+
+class Latex(commands.Cog):
+    """Renders latex."""
+
+    @commands.command()
+    @commands.max_concurrency(1, commands.BucketType.guild, wait=True)
+    async def latex(self, ctx: commands.Context, *, query: str) -> None:
+        """Renders the text in latex and sends the image."""
+        query = _prepare_input(query)
+        query_hash = hashlib.md5(query.encode()).hexdigest()
+        image_path = CACHE_DIRECTORY / f"{query_hash}.png"
```

Comment:
I think you meant to do ```py
image_path = f"CACHE_DIRECTORY/{query_hash}.png"```. Right now, you're dividing 2 strings by each other, which will cause an exception.

Reply:
As per https://github.com/python-discord/sir-lancebot/blob/89ebfc684d257286bc7cfcb9023658952314cd3b/bot/exts/fun/latex/_latex_cog.py#L21-L23

`CACHE_DIRECTORY` is a [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path) instance, which supports the dividing operator.

I could not find the documentation for this, but the source for this is here: https://github.com/python/cpython/blob/2e6798f35260ff90129861ef1f289ac40c0396c2/Lib/pathlib.py#L851-L861

Reply:
If you're curious, the docs are the first entry under [the operators heading](https://docs.python.org/3/library/pathlib.html#operators),

Reply:
Thanks for the info!